catroid/src/main/java/org/catrobat/catroid/content/WhenCollideWithScript.java

```diff
@@ -0,0 +1,163 @@
+ * http://developer.catrobat.org/license_additional_term
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.catrobat.catroid.content;
+
+import android.content.Context;
+
+import org.catrobat.catroid.ProjectManager;
+import org.catrobat.catroid.R;
+import org.catrobat.catroid.content.bricks.Brick;
+import org.catrobat.catroid.content.bricks.ConcurrentFormulaHashMap;
+import org.catrobat.catroid.content.bricks.ScriptBrick;
+import org.catrobat.catroid.content.bricks.WhenCollideWithBrick;
+import org.catrobat.catroid.content.eventids.EventId;
+import org.catrobat.catroid.content.eventids.WhenConditionEventId;
+import org.catrobat.catroid.formulaeditor.Formula;
+import org.catrobat.catroid.formulaeditor.FormulaElement;
+import org.catrobat.catroid.formulaeditor.Sensors;
+
+public class WhenCollideWithScript extends Script {
+
+	private enum CollisionType {
+		FINGER,
+		EDGE,
+		SPRITE
+	}
+
+	private static final long serialVersionUID = 1L;
+	private String collisionObjectName;
+	private Sprite collisionObject;
+
+	private ConcurrentFormulaHashMap formulaMap = new ConcurrentFormulaHashMap();
+
+	public WhenCollideWithScript() {
+		this.collisionObject = getBackgroundSprite();
+		if (this.collisionObject != null) {
+			formulaMap.putIfAbsent(Brick.BrickField.IF_CONDITION,
+					new Formula(getFormulaElement(CollisionType.SPRITE,
+							this.collisionObject.getName())));
+		}
+		this.collisionObjectName = null;
+	}
+
+	@Override
+	public ScriptBrick getScriptBrick() {
+		if (scriptBrick == null) {
+			scriptBrick = new WhenCollideWithBrick(this);
+		}
+		return scriptBrick;
+	}
+
+	public String getSpriteToCollideWithName() {
+		return collisionObject != null ? collisionObject.getName() : collisionObjectName;
+	}
+
+	@Override
+	public Script clone() throws CloneNotSupportedException {
+		WhenCollideWithScript clone = (WhenCollideWithScript) super.clone();
+		clone.formulaMap = formulaMap.clone();
+		return clone;
+	}
+
+	@Override
+	public void addRequiredResources(Brick.ResourcesSet requiredResourcesSet) {
+		for (Formula formula : formulaMap.values()) {
+			formula.addRequiredResources(requiredResourcesSet);
+		}
+		for (Brick brick : brickList) {
+			brick.addRequiredResources(requiredResourcesSet);
+		}
+	}
+
+	public ConcurrentFormulaHashMap getFormulaMap() {
+		return formulaMap;
+	}
+
+	public Sprite getSpriteWithName(String name) {
+		Scene thisScene = ProjectManager.getInstance().getCurrentlyEditedScene();
+
+		return thisScene != null ? thisScene.getSprite(name) : null;
+	}
+
+	public Sprite getBackgroundSprite() {
```

Comment:
This method is only used internally within the class. Please mark it (and others like it) as private.